queries/djot/highlights.scm

```diff
@@ -0,0 +1,276 @@
+
+(heading5) @markup.heading.5
+
+(heading6) @markup.heading.6
+
+(thematic_break) @string.special
+
+[
+  (div_marker_begin)
+  (div_marker_end)
+] @punctuation.delimiter
+
+[
+  (code_block)
+  (raw_block)
+  (frontmatter)
+] @markup.raw
+
+; Remove @markup.raw for code with a language spec
+(code_block
+  .
+  (code_block_marker_begin)
+  (language)
+  (code) @none)
+
+[
+  (code_block_marker_begin)
+  (code_block_marker_end)
+  (raw_block_marker_begin)
+  (raw_block_marker_end)
+] @punctuation.delimiter
+
+(language) @tag.attribute
+
+(language_marker) @tag.delimiter
+
+(class_name) @tag.attribute
+
+(block_quote) @markup.quote
+
+(block_quote_marker) @punctuation.special
+
+(table_header) @markup.heading
+
+(table_header
+  "|" @punctuation.special)
+
+(table_row
+  "|" @punctuation.special)
+
+(table_separator) @punctuation.special
+
+(table_caption
+  (marker) @punctuation.special)
+
+(table_caption) @markup.heading
+
+[
+  (list_marker_dash)
+  (list_marker_plus)
+  (list_marker_star)
+  (list_marker_definition)
+  (list_marker_decimal_period)
+  (list_marker_decimal_paren)
+  (list_marker_decimal_parens)
+  (list_marker_lower_alpha_period)
+  (list_marker_lower_alpha_paren)
+  (list_marker_lower_alpha_parens)
+  (list_marker_upper_alpha_period)
+  (list_marker_upper_alpha_paren)
+  (list_marker_upper_alpha_parens)
+  (list_marker_lower_roman_period)
+  (list_marker_lower_roman_paren)
+  (list_marker_lower_roman_parens)
+  (list_marker_upper_roman_period)
+  (list_marker_upper_roman_paren)
+  (list_marker_upper_roman_parens)
+] @markup.list
+
+(list_marker_task
+  (unchecked) @constant.builtin) @markup.list.unchecked
+
+(list_marker_task
+  (checked) @constant.builtin) @markup.list.checked
+
+[
+  (ellipsis)
+  (en_dash)
+  (em_dash)
+  (straight_quote)
+] @string.special
+
+(frontmatter_marker) @punctuation.delimiter
+
+(emphasis) @markup.italic
+
+(strong) @markup.strong
+
+(emphasis
+  (emphasis_begin) @punctuation.delimiter)
+
+(emphasis
+  (emphasis_end) @punctuation.delimiter)
+
+(strong
+  (strong_begin) @punctuation.delimiter)
+
+(strong
+  (strong_end) @punctuation.delimiter)
+
+(highlighted
+  [
+    "{="
+    "=}"
+  ] @punctuation.delimiter)
+
+(insert
+  [
+    "{+"
+    "+}"
+  ] @punctuation.delimiter)
+
+(delete
+  [
+    "{-"
+    "-}"
+  ] @punctuation.delimiter)
+
+(superscript
+  [
+    "^"
+    "{^"
+    "^}"
+  ] @punctuation.delimiter)
+
+(subscript
+  [
+    "~"
+    "{~"
+    "~}"
+  ] @punctuation.delimiter)
+
+(verbatim) @markup.raw
+
+[
+  (verbatim_marker_begin)
+  (verbatim_marker_end)
+] @punctuation.delimiter
+
+(math) @markup.math
+
+[
+  (math_marker)
+  (math_marker_begin)
+  (math_marker_end)
+] @punctuation.delimiter
+
+(raw_inline) @markup.raw
+
+[
+  (raw_inline_attribute)
+  (raw_inline_marker_begin)
+  (raw_inline_marker_end)
+] @punctuation.delimiter
+
+[
+  "{"
+  "}"
+  "!["
+  "["
+  "]"
+  "("
+  ")"
+  "<"
+  ">"
+] @punctuation.bracket
+
+(comment) @comment
+
+(class) @tag.attribute
```

Comment:
If this and `(class_name)` are CSS classes they should use `@type`.

Reply:
Hey @treeman, are these for css classes?

Reply:
Yeah they usually refer to css classes (although Djot can generate other formats than html, we should probably have that as the guiding star).

With regards to the queries, I just picked some that looked decent to me, so feel free to change them to something more deliberate :)

Reply:
Oh, we should probably simplify the grammar to merge `(class)` and `(class_name)` as well.

Reply:
class can be the class name + the block, maybe